tests/checkup_test.go

```diff
@@ -0,0 +1,237 @@
+/*
+ * This file is part of the kiagnose project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2023 Red Hat, Inc.
+ *
+ */
+
+package tests
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gstruct"
+
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	testServiceAccountName                     = "rt-checkup-sa"
+	testKiagnoseConfigMapAccessRoleName        = "kiagnose-configmap-access"
+	testKiagnoseConfigMapAccessRoleBindingName = testKiagnoseConfigMapAccessRoleName
+	testConfigMapName                          = "rt-checkup-config"
+	testCheckupJobName                         = "rt-checkup"
+)
+
+var _ = Describe("Execute the checkup Job", func() {
+	var checkupServiceAccount *corev1.ServiceAccount
+	var kiagnoseConfigMapAccessRole *rbacv1.Role
+	var kiagnoseConfigMapAccessRoleBinding *rbacv1.RoleBinding
+	var configMap *corev1.ConfigMap
+	var checkupJob *batchv1.Job
+
+	BeforeEach(func() {
+		var err error
+
+		checkupServiceAccount = newServiceAccount()
+		_, err = client.CoreV1().ServiceAccounts(testNamespace).Create(context.Background(), checkupServiceAccount, metav1.CreateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		kiagnoseConfigMapAccessRole = newKiagnoseConfigMapAccessRole()
+		_, err = client.RbacV1().Roles(testNamespace).Create(context.Background(), kiagnoseConfigMapAccessRole, metav1.CreateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		kiagnoseConfigMapAccessRoleBinding = newKiagnoseConfigMapAccessRoleBinding()
+		_, err = client.RbacV1().RoleBindings(testNamespace).Create(
+			context.Background(),
+			kiagnoseConfigMapAccessRoleBinding,
+			metav1.CreateOptions{},
+		)
+		Expect(err).NotTo(HaveOccurred())
+
+		configMap = newConfigMap()
+		_, err = client.CoreV1().ConfigMaps(testNamespace).Create(context.Background(), configMap, metav1.CreateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		checkupJob = newCheckupJob()
+		_, err = client.BatchV1().Jobs(testNamespace).Create(context.Background(), checkupJob, metav1.CreateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		DeferCleanup(func() {
```

Comment:
I do not think this block is correct.
If a creation of an object was successful, then you need to use a deferred cleanup.
Not group them all at the end.

As an example, if the job create failed, no one will cleanup the configmap and RBAC items.

Reply:
Added DeferCleanup() after each successful object creation.